specification/network/resource-manager/Microsoft.Network/stable/2018-06-01/networkWatcher.json

```diff
@@ -3148,6 +3273,221 @@
         }
       },
       "description": "Connection state snapshot."
+    },
+    "NetworkConfigurationDiagnosticParameters": {
+      "description": "Parameters to get network configuration diagnostic.",
+      "required": [
+        "targetResourceId",
+        "queries"
+      ],
+      "properties": {
+        "targetResourceId": {
+          "type": "string",
+          "description": "The ID of the target resource to perform network configuration diagnostic. Valid options are VM, NetworkInterface, VMSS/NetworkInterface and Application Gateway."
+        },
+        "queries": {
+          "type": "array",
+          "description": "List of traffic queries.",
+          "items": {
+            "$ref": "#/definitions/TrafficQuery"
+          }
+        }
+      }
+    },
+    "TrafficQuery": {
+      "description": "Parameters to compare with network configuration.",
+      "required": [
+        "direction",
+        "protocol",
+        "source",
+        "destination",
+        "destinationPort"
+      ],
+      "properties": {
+        "direction": {
+          "type": "string",
+          "enum": [
+            "Inbound",
+            "Outbound"
+          ],
+          "x-ms-enum": {
+            "name": "Direction",
+            "modelAsString": true
+          },
+          "description": "The direction of the traffic. Accepted values are 'Inbound' and 'Outbound'."
+        },
+        "protocol": {
+          "type": "string",
+          "description": "Protocol to be verified on. Accepted values are '*', TCP, UDP."
+        },
+        "source": {
+          "type": "string",
+          "description": "Traffic source. Accepted values are '*', IP Address/CIDR, Service Tag."
+        },
+        "destination": {
+          "type": "string",
+          "description": "Traffic destination. Accepted values are: '*', IP Address/CIDR, Service Tag."
+        },
+        "destinationPort": {
+          "type": "string",
+          "description": "Traffice destination port. Accepted values are '*', port (for example, 3389) and port range (for example, 80-100)."
+        }
+      }
+    },
+    "NetworkConfigurationDiagnosticResponse": {
+      "description": "Results of network configuration diagnostic on the target resource.",
+      "properties": {
+        "results": {
```

Comment:
"value" array is the standard here.

Reply:
It's not ObjectListPage. Value would be there only in this case...